enarx-keep-sgx-shim/src/start.S

```diff
@@ -142,6 +169,12 @@ enclave:
 .Lsyscall:
     movq    $0,                     SRSP(%r11)      # Clear syscall return stack pointer field
     mov     %rax,                   %rsp            # Restore the syscall return stack pointer
+
+    movq    $~0,                    %rdx            # Set mask for xrstor in rdx
+    movq    $~0,                    %rax            # Set mask for xrstor in rax
+    xrstor  (%rsp)                                  # Restore extended CPU state
+    mov     %r8,                    %rsp            # Restore stack alignment
```

Comment:
is `%r8` saved somewhere?

Reply:
This is a good question. I'm not sure how to tell which registers are guaranteed to not be used during the syscall. Any advice? Or maybe I can put the old `%rsp` on the newly-aligned stack?

Reply:
@haraldh Sorry, maybe you are asking for context. The pre-alignment `%rsp` is saved in `%r8` before the syscall and I'm only assuming it will still be there after the syscall.

Reply:
The value is saved here: https://github.com/enarx/enarx/pull/427/files#diff-c44d5e117cccb73d5356de2ed1a1b8a2R207

Reply:
@haraldh I see you're using `%rbx` [here](https://github.com/haraldh/enarx/blob/sev_make/enarx-keep-sev-shim/src/arch/x86_64/asm/syscall.s#L27) - maybe that one is better in this case too.

Reply:
I don't know.. just asking, if `%r8` has to be pushed on `%rsp` later also, or if that is the case anyway.

Reply:
I mean, if it is saved and restored on enclave exit/enter anyway.